Sources/UnfairLock.swift

```diff
@@ -0,0 +1,29 @@
+import os.lock
+
+final class UnfairLock {
+    private let lockPointer: os_unfair_lock_t
+
+    init() {
+        self.lockPointer = .allocate(capacity: 1)
+        self.lockPointer.initialize(to: os_unfair_lock())
+    }
+
+    deinit {
+        self.lockPointer.deinitialize(count: 1)
+        self.lockPointer.deallocate()
+    }
+
+    func lock() {
+        os_unfair_lock_lock(self.lockPointer)
+    }
+
+    func unlock() {
```

Comment:
[detail] Would prefer to keep lock and unlock functions private for now, and only expose them when needed